### PR TITLE
Add `JsonArtifact`

### LINKED ISSUE #935: Add `JsonArtifact`
- [x] I have read and agree to the [contributing guidelines](https://github.com/griptape-ai/griptape#contributing).

**Is your feature request related to a problem? Please describe.**
I want a way to pass around json-serialized values without having to explicitly de/serialize the dict myself

**Describe the solution you'd like**
`JsonArtifact` that takes a dict and serializes to json.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by vachillo**
- [x] I have read and agree to the contributing guidelines for [submitting new pull requests](https://github.com/griptape-ai/griptape?tab=readme-ov-file#submitting-pull-requests).

## Describe your changes
- adding JsonArtifact for de/serializing values automatically

## Issue ticket number and link
closes #935 

----
📚 Documentation preview 📚: https://griptape--937.org.readthedocs.build//937/

### CODE REVIEW

**Comment by collindutter** [comment]
Could you throw in a `JsonLoader` to knock out https://github.com/griptape-ai/griptape/issues/439?

File: griptape/artifacts/json_artifact.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    from griptape.artifacts import BaseArtifact
+
+
+@define
+class JsonArtifact(TextArtifact):
+    value: Any = field(converter=json.dumps, metadata={"serializable": True})  # pyright: ignore reportRedeclaration
+
+    @property
+    def value(self) -> Any:
+        return json.loads(self.value)
```
Comment (collindutter) [inline griptape/artifacts/json_artifact.py:17]: The codecov warning here seems worth testing.

File: pyproject.toml [hunk 2]
```diff
@@ -28,6 +28,7 @@ stringcase = "^1.2.0"
 docker = "^7.1.0"
 sqlalchemy = "~=1.0"
 requests = "^2.32.0"
+jsonalias = "^0.1.2"
```
Comment (collindutter) [inline pyproject.toml:31]: I'm not sure this is even worth including since its usage is so limited. Can you just include the type in `JsonArtifact`?

File: tests/unit/artifacts/test_json_artifact.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import json
+import pytest
+from griptape.artifacts import JsonArtifact, TextArtifact
+
+
+class TestJsonArtifact:
+    def test_value_type_conversion(self):
+        assert JsonArtifact({"foo": "bar"}).value == json.loads(json.dumps({"foo": "bar"}))
+        string = json.dumps({"foo": {}})
+        print(string)
```
Comment (collindutter) [inline tests/unit/artifacts/test_json_artifact.py:10]: Remove

File: tests/unit/artifacts/test_json_artifact.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        string = json.loads(string)
+        print(string)
```
Comment (collindutter) [inline tests/unit/artifacts/test_json_artifact.py:12]: Remove

File: docs/griptape-framework/data/artifacts.md [hunk 5]
```diff
@@ -58,3 +58,7 @@ A [BooleanArtifact](../../reference/griptape/artifacts/boolean_artifact.md) is u
 A [GenericArtifact](../../reference/griptape/artifacts/generic_artifact.md) can be used as an escape hatch for passing any type of data around the framework.
 It is generally not recommended to use this Artifact type, but it can be used in a handful of situations where no other Artifact type fits the data being passed.
 See [talking to a video](../../examples/talk-to-a-video.md) for an example of using a `GenericArtifact` to pass a Gemini-specific video file.
+
+## JsonArtifact
```
Comment (collindutter) [inline docs/griptape-framework/data/artifacts.md:62]: Header should just be `Json`

File: griptape/artifacts/json_artifact.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+from griptape.artifacts import BaseArtifact
+
+Json = Union[dict[str, "Json"], list["Json"], str, int, float, bool, None]
+
+
+@define
+class JsonArtifact(BaseArtifact):
+    _obj: Any = field(metadata={"serializable": True}, alias="obj")
+    value: dict = field(init=False, metadata={"serializable": True})
```
Comment (collindutter) [inline griptape/artifacts/json_artifact.py:16]: Should be `value: Json`?

File: griptape/artifacts/json_artifact.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    _json_str: str = field(init=False, metadata={"serializable": True})
+
+    def __attrs_post_init__(self) -> None:
+        if self._obj is None:
+            self._obj = {}
+        self._json_str = json.dumps(self._obj)
```
Comment (collindutter) [inline griptape/artifacts/json_artifact.py:22]: Why do we need a field for this? Can we just dumps in `to_text`?

File: griptape/artifacts/json_artifact.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+from attrs import define, field
+
+from griptape.artifacts import BaseArtifact
+
+Json = Union[dict[str, "Json"], list["Json"], str, int, float, bool, None]
+
+
+@define
+class JsonArtifact(BaseArtifact):
+    _obj: Any = field(metadata={"serializable": True}, alias="obj")
```
Comment (collindutter) [inline griptape/artifacts/json_artifact.py:15]: What does this field provide?

File: tests/unit/artifacts/test_json_artifact.py [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+        new = JsonArtifact({"foo": "bar"}) + JsonArtifact({"value": "baz"})
+        assert new.value == {"foo": "bar", "value": "baz"}
+
+        assert (JsonArtifact({"foo": "bar"}) + JsonArtifact({"foo": "baz"})).value == JsonArtifact({"foo": "baz"}).value
+        with pytest.raises(ValueError):
+            JsonArtifact({"foo": "bar"}) + TextArtifact("invalid json")
+
+    def test_to_text(self):
+        assert JsonArtifact({"foo": "bar"}).to_text() == json.dumps({"foo": "bar"})
```
Comment (collindutter) [inline tests/unit/artifacts/test_json_artifact.py:25]: Can you please add tests for all the types that `Json` can be? dict[str, "Json"], list["Json"], str, int, float, bool, None

**Review by collindutter — CHANGES_REQUESTED** [review]

File: griptape/artifacts/json_artifact.py (hunk 8 above)
Comment (vachillo) [reply griptape/artifacts/json_artifact.py:15]: this is just the original object being converted. different than `value`.

**Review by collindutter — DISMISSED** [review]

**Review by collindutter — APPROVED** [review]
